public/model/customLayerFunctions.ts

```diff
@@ -65,15 +65,17 @@ const addNewLayer = (
         id: layerConfig.id,
         type: 'raster',
         source: layerConfig.id,
+        paint: {
+          'raster-opacity': layerConfig.opacity / 100,
+        },
+        layout: {
+          visibility: layerConfig.visibility === 'visible' ? 'visible' : 'none',
+        },
+        minzoom: layerConfig.zoomRange[0],
```

Comment:
Why not just create layer here and call update to update everything else? This will keep all our layer properties update in one place.

Reply:
This function `addNewLayer`  will only be called when initially render layer which need all properties together, so that I think here combine all properties together. If it need update the some property, it will call another function `updateLayerConfig`. 

https://github.com/opensearch-project/dashboards-maps/blob/3b8396ce96d4a5f2b106189b5a6ea2dbfca81867/public/model/customLayerFunctions.ts#L47

https://github.com/opensearch-project/dashboards-maps/blob/3b8396ce96d4a5f2b106189b5a6ea2dbfca81867/public/model/customLayerFunctions.ts#L10

Reply:
Yes. Now you have to maintain the property change in two places. I am proposing something like below

```
addNewLayer = (props) => {

add source
add layer
updateLayer(props)
}
```
In future if we add new properties, we don't have to update in two places.

Reply:
Looks good, I think we can do that later.

Reply:
Please create an issue if we want to do this later.